fdbclient/DatabaseContext.h

```diff
@@ -78,20 +69,19 @@ class DatabaseContext : public ReferenceCounted<DatabaseContext>, public FastAll
 	
 	void setOption( FDBDatabaseOptions::Option option, Optional<StringRef> value );
 
-	Error deferredError;
-	bool lockAware;
+	bool isError() const;
 
-	bool isError() {
-		return deferredError.code() != invalid_error_code;	
-	}
+	void checkDeferredError() const;
 
-	void checkDeferredError() {
-		if(isError()) {
-			throw deferredError;
-		}
-	}
+	int apiVersionAtLeast(int minVersion) const;
 
-	int apiVersionAtLeast(int minVersion) { return apiVersion < 0 || apiVersion >= minVersion; }
+	void validateVersion(Version version) const;
+	TaskPriority taskID() const;
+	bool switchable() const;
+	bool& enableLocalityLoadBalance();
```

Comment:
This is more a style question, but for these getter functions, are there any opinions on these names vs. ones like `getTaskID()`, `isSwitchable()`, etc? I'm also a little unaccustomed to seeing mutations done this way (e.g. `apiVersion() = 100`). I'm fine with this style if others are.

This particular field (`enableLocalityLoadBalance`) is perhaps misleading, though, when converted to a function, as it sounds as if it has some side effect of enabling a feature. We may want to rename it to something like `localityLoadBalanceEnabled`.

Reply:
Returning a reference to do something like `apiVersion() = 100` is what allows me to hide the implementation in a pimpl. Using getters and setters would also work, but it is more code and I prefer it this way (though if this is just me I am fine changing it).

I agree with the comment about `enableLocalityLoadBalance` - I simply didn't rename anything and instead just translated fields to methods... I will change this name.

Reply:
I have a slight preference for the getter/setter approach, but not to the extent that I would attempt to block this style if we're happy with it. Not sure what the right way to determine that is, or if it's even worth the effort.